### PR TITLE
Feature/reorganise assets and layouts

### LINKED ISSUE #21: better error pages
the number is not the important information

### LINKED ISSUE #30: Better organisation of Stylesheets
@jordanmaguire, @Julia-Alison and myself had a long discussion about how we will rearrange/organise the stylesheets and layouts.

This is the outcome:

img_2679

Will be done asap.

### LINKED ISSUE #33: Styles for selected menu option in admin page

### LINKED ISSUE #38: Home link does not go to home
Much disappoint.

### LINKED ISSUE #40: JS should not be loaded in header

### PR DESCRIPTION
**Pull request by pzi**
# Reorganise all the styles, js and layout html

√ Fixes #30 
√ Fixes #21 
√ Fixes #33 
√ Fixes #38 
√ Fixes #40

### CODE REVIEW

File: app/assets/stylesheets/_global.sass [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Shared Styles for the entire webapp
+
+@import "normalize-rails"
+@import "bourbon"
+
+/*!
+ * Author:       The Frontier Group
+ * Author URI:   http://www.thefrontiergroup.com.au/
+ */
```
Comment (osiro) [inline app/assets/stylesheets/_global.sass]: Is this information important for some reason?

File: app/views/devise/registrations/edit.html.haml [hunk 2]
```diff
@@ -7,7 +7,7 @@
     = f.input :email, autofocus: true
     - if devise_mapping.confirmable? && resource.pending_reconfirmation?
       %p Currently waiting confirmation for: #{resource.unconfirmed_email}
-    = f.input :password, autocomplete: "off", hint: "leave blank if you don't want to change it"
+    = f.input :password, autocomplete: "off", hint: "Minimum is 8 characters. Leave blank if you don't want to change it."
```
Comment (osiro) [inline app/views/devise/registrations/edit.html.haml]:
```ruby
"Minimum is #{resource.class.password_length.min} characters. ...."
```

File: app/views/devise/registrations/new.html.haml [hunk 3]
```diff
@@ -5,7 +5,7 @@
   = simple_form_for(resource, as: resource_name, url: registration_path(resource_name), html: { class: "authentication-form" }) do |f|
 
     = f.input :email, autofocus: true
-    = f.input :password, autocomplete: "off"
+    = f.input :password, autocomplete: "off", hint: "Minimum is 8 characters"
```
Comment (osiro) [inline app/views/devise/registrations/new.html.haml]: Same as above

File: app/views/layouts/admin/_sidebar_links.html.haml [hunk 4]
```diff
@@ -1 +1,2 @@
-%li= link_to 'Users', admin_users_path
\ No newline at end of file
+-# Hello Jordan, I'd be super keen to make this better :) Will probably be some cool helper
+%li{class: ("active" if current_page?(admin_users_path))}= link_to 'Users', admin_users_path
```
Comment (osiro) [inline app/views/layouts/admin/_sidebar_links.html.haml]: You've got some alternatives, moving to a helper would be a good one.

```ruby
def menu_link_to(content, path)
  css_class = current_page?(admin_users_path) ? "active" : ""
  content_tag(:li, class: css_class) do
    link_to content, path
  end
end
```

File: app/views/layouts/admin/_admin_navigation.html.haml [hunk 5]
```diff
@@ -0,0 +1 @@
+%li= link_to "Dashboard", admin_dashboard_index_path
```
Comment (jordanmaguire) [inline app/views/layouts/admin/_admin_navigation.html.haml]: No need to have `admin/_admin_navigation`, can just do `admin/_navigation`

File: app/views/layouts/admin/_sidebar_links.html.haml (hunk 4 above)
Comment (jordanmaguire) [inline app/views/layouts/admin/_sidebar_links.html.haml]: I will tackle this.

File: app/views/layouts/member/_member_navigation.html.haml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+%li= link_to "Dashboard", member_dashboard_index_path
```
Comment (jordanmaguire) [inline app/views/layouts/member/_member_navigation.html.haml]: `member/_navigation`

File: app/assets/stylesheets/_global.sass (hunk 1 above)
Comment (pzi) [inline app/assets/stylesheets/_global.sass]: It's called a banner and I'd like that to be on top of all our CSS. Every library/plugin these days has one and it's an easy and "nice" way to credit us unobtrusively.

File: app/assets/stylesheets/global/_base.sass [hunk 7]
```diff
@@ -1,28 +1,23 @@
...
-
-.container
-  padding: $gutter
-
 a
   color: $link-color
+  text-decoration: none
+  +transition(color 150ms ease)
   &:hover
     color: $link-hover-color
```
Comment (Julia-Alison) [inline app/assets/stylesheets/global/_base.sass]: i'd nuke this, just use darken($link-color, 0.15)

**Comment by Julia-Alison** [comment]
:cactus: good

**Pull request merged**